File: src/main/kotlin/features/toggles/toggleUtils.kt
package features.toggles

import features.chrome.getStoredValue
import kotlin.js.Promise

fun getStoredToggle(name: String): Promise<Boolean> {
    return Promise { resolve, reject ->
        getStoredValue(name).then<dynamic> {
            if (it is Boolean) {
                resolve(it)
            } else {
                reject(Error("Toggle with $name is not a boolean"))
            }
        }
    }
}

File: src/features/toggles/toggleUtils.ts
import { getStoredValue, setStoredValue } from "../chrome/chromeUtils";

import { ChangeEvent } from "react";

export async function getStoredToggle(name: string): Promise<boolean> {
  return getStoredValue(name) as Promise<boolean>;
}

export function toggleChangeHandler(event: ChangeEvent<HTMLInputElement>) {
  setStoredValue(event.target.id, event.target.checked);
}
